### ISSUE TITLE
Strike-through accepts large and mismatching numbers of tildes for prefix and suffix

### ISSUE THREAD

**Issue by Crissov** [github/cmark#118]
Everybody agreed in #71 and #99 that double tilde should be the recommended way to mark up stricken text `~~foo~~`, although cmark-gfm still needs to support single tilde markers `~foo~` for legacy reasons. However, cmark-gfm also supports triple tildes and really any number of matching tildes `~~~foo~~~` and, what's worse, even any mismatching numbers of tildes before and after the stricken text `~~~foo~`, `~foo~~~`. I doubt this is necessary for much existing content. Therefore:

1. Require the same number of tildes on both sides.
2. Limit this number to 2, keeping legacy support for just 1.

This would solve the (admittedly rare) problem of fenced code blocks being not recognized. commonmark/CommonMark#537

```` markdown
~~~foo~~~
should perhaps be a code block
~~~
````
```` markdown
~~~bar~
should definitely be a code block
~~~
````

**Comment by kivikakk** [comment]
You're right — our old parser did not support strikethroughs in number greater than 2. I'll implement this.

**Comment by Crissov** [comment]
Will you also fix `~~foo~` and  `~foo~~`?

**Comment by kivikakk** [comment]
That is indeed fixed in #120.
